### DISCUSSION TITLE
Handling External Events in NextUI Button Components with DnD Kit

### DISCUSSION THREAD

**Discussion by masanori-iwata** [heroui-inc/heroui#2241] [Q&A]
Hello,

I am currently working on a project where I'm trying to create a sortable list using NextUI and DnD Kit. However, I've encountered an issue where the NextUI button components do not seem to accept or respond to the events generated by DnD Kit.

I've tried a few workarounds but have not been successful in integrating the external events from DnD Kit into the NextUI button components. I believe this might be an issue related to event handling or propagation in the NextUI components when used with DnD Kit.

I am reaching out to the community for any insights or suggestions on how to resolve this issue. Has anyone else faced a similar challenge? How did you manage to integrate external event handling, specifically from DnD Kit, into NextUI components?

Any guidance or examples would be greatly appreciated.

Thank you!

**Comment by MarcusDelvecchio** [comment]
having a similar issue with NextUI elements (Popover and Button) interfering with DnD Kit draggable. Interested in whether any progress has been made here.

**Comment by madb88** [comment]
I had a similar problem where the button wasn’t responding to being dragged, but after adding the draggable option to the button, it started working. Shadcn ui buttons worked without it.

`<Button
							isIconOnly
							className="cursor-move"
							color={"secondary"}
							type={"button"}
							draggable
							{...attributes}
							{...listeners}
						>
							<Grip />
						</Button>`
